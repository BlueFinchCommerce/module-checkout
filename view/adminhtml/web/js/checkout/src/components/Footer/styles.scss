@import '@/styles/global/_breakpoints';

.footer-container {
  display: var(--footer-display);
  background: var(--footer-background);
  border-top: var(--footer-border-top);
  padding: var(--footer-container-mobile-paddings, 20px 20px);

  .footer-block {
    padding: 0 0 var(--indent__xxl);
    margin: 0 auto;
    display: flex;
    align-items: baseline;
    flex-wrap: wrap;
  }

  p {
    color: var(--copyright-text-color, var(--white__color));
    max-width: var(--layout__max-width);
    padding: 5px var(--indent__base);
    margin: var(--footer-mobile-container-margin, 0 auto);
    font-size: var(--footer-copy-font-size, 15px);
    font-family: var(--font-family-copy);
  }
}

@mixin largeWidth() {
  .footer-container {
    padding: var(--indent__base);

    p {
      margin: var(--footer-container-margin, 0 auto);
    }

    .footer-block {
      justify-content: var(--footer-block-justify-content-desktop, initial);
      padding: var(--footer-paddings, 0);

      p {
        margin: var(--footer-desktop-container-margin, 0 0 0 auto);
      }
    }
  }
}

.mock-laptop,
.mock-desktop {
    @include largeWidth;
}

@media screen and (min-width: $screen__m) {
  .mock-full {
      @include largeWidth;
  }
}
