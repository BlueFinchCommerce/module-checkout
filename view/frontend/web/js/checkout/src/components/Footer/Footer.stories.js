import FooterStoryBook from './FooterStoryBook/FooterStoryBook.vue';

export default {
  title: 'Checkout Structure Components/App/Footer',
  component: FooterStoryBook,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
};

const Template = (args) => ({
  components: { FooterStoryBook },
  setup() {
    return { args };
  },
  template: '<FooterStoryBook v-bind="args"/>',
});

export const FooterComponent = Template.bind({});

FooterComponent.args = {
  copyrightText: '@COPYRIGHT 2024',
};
