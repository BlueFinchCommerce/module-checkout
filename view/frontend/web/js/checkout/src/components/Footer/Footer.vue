<template>
  <footer
    class="footer-container"
    :style="style"
  >
    <div class="footer-block">
      <FooterIcons />
      <TextField
        :data-cy="'copyright-text'"
        :text="copyrightText"
      />
    </div>
  </footer>
</template>

<script>
import { reactive, computed } from 'vue';
import { mapState } from 'pinia';
import TextField from '@/components/Core/ContentComponents/TextField/TextField.vue';
import FooterIcons from '@/components/Footer/FooterIcons/FooterIcons.vue';
import useConfigStore from '@/stores/ConfigStores/ConfigStore';

export default {
  name: 'AppFooter',
  components: {
    TextField,
    FooterIcons,
  },
  props: {
    backgroundColor: {
      type: String,
    },
  },
  computed: {
    ...mapState(useConfigStore, ['copyrightText']),
  },
  setup(props) {
    const reactiveProps = reactive(props);
    return {
      style: computed(() => ({
        backgroundColor: reactiveProps.backgroundColor,
      })),
    };
  },
};
</script>

<style lang="scss" scoped>
@import "@/components/Footer/styles.scss";
</style>
