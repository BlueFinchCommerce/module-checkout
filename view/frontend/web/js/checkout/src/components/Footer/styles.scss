@import '@/styles/global/_breakpoints';

.footer-container {
  display: var(--footer-display);
  background: var(--footer-background);
  border-top: var(--footer-border-top);
  padding: var(--footer-container-mobile-paddings, 20px 20px);

  .footer-block {
    margin: 0 auto;
    display: flex;
    align-items: baseline;
    flex-wrap: wrap;
  }

  p {
    color: var(--text__color);
    max-width: var(--layout__max-width);
    padding: 5px var(--indent__base);
    margin: var(--footer-mobile-container-margin, 0 auto);
    font-size: var(--footer-copy-font-size, 15px);
    font-family: var(--font-family-copy);
  }
}

@media (min-width: $screen__m) {
  .footer-container {
    padding: var(--indent__base);

    p {
      margin: var(--footer-container-margin, 0 auto);
    }

    .footer-block {
      justify-content: var(--footer-block-justify-content-desktop, initial);
      padding: var(--footer-paddings, 0);
    }
  }
}
