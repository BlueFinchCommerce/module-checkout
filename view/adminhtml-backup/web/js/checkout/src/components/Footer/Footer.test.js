import { shallowMount } from '@vue/test-utils';
import { setActivePinia, createPinia } from 'pinia';
import Footer from './Footer.vue';

describe('Footer.vue', () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it('Footer exist', () => {
    const wrapper = shallowMount(Footer);
    expect(wrapper.findComponent(Footer).exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
