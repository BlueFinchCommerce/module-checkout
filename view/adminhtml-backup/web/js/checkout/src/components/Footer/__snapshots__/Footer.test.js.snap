// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue Footer exist 1`] = `
<footer
  class="footer-container"
>
  <div
    class="footer-block"
  >
    <footer-icons-stub />
    <text-field-stub
      text=""
    />
  </div>
</footer>
`;
